tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
```

Comment:
I think you can drop this loop , instead of extracting values from hash to variable you can just use it directly from hash